src/app/shared/components/guided-tour/dialog/service.ts

```diff
@@ -0,0 +1,190 @@
+// Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Injectable, Injector, ComponentRef, Renderer2, RendererFactory2} from '@angular/core';
+import {Overlay, OverlayConfig, OverlayRef, ConnectionPositionPair} from '@angular/cdk/overlay';
+import {ComponentPortal, PortalInjector} from '@angular/cdk/portal';
+import {DialogComponent} from './component';
+import {DialogRef} from './overlay-ref';
+import {DialogConfig, DEFAULT_CONFIG} from './entity';
+import {DialogHelperService} from './helper.service';
+import {GuidedTourStep, STEP_DATA, STEP_ORDER} from '../entity';
+import {POSITION_MAP} from '../utils';
+
+@Injectable()
+export class DialogService {
+  private _tempSteps: GuidedTourStep[] = [];
+  private _renderer: Renderer2;
+
+  dialogRef: DialogRef;
```

Comment:
Due to using dialog as a base for all "popups" there are lots of workarounds to avoid some of the logic applied by the dialog. Querying for `km-main`, removing/applying the backdrop classes manually, etc. 

I am not sure if the dialog is needed at all as a base here. Since custom backdrop has to be applied manually anyway we also don't really need overlay container manipulation that much.

Tooltips i.e. are displayed without so much custom logic and using tooltip directive is so much easier. I guess we have to reevaluate how complex offered tours should be. Right now it feels like there is too much going on underneath (hacky solutions) and the user experience does not benefit from that very much.

Reply:
Um...maybe? Sorry don't know a proper answer, we might have to figure that out ^^